packages/react-devtools-scheduling-profiler/src/import-worker/preprocessData.js

```diff
@@ -656,6 +676,12 @@ export default function preprocessData(
 
   timeline.forEach(event => processTimelineEvent(event, profilerData, state));
 
+  if (profilerVersion === null) {
+    throw new InvalidProfileError(
+      `This version of profiling data is not supported by the current profiler.`,
+    );
+  }
+
   // Validate that all events and measures are complete
   const {measureStack} = state;
   if (measureStack.length > 0) {
```

Comment:
just curious, when does the code in this `preprocessData` module end up running? (happy to chat offline about it too)

Reply:
Looks like it runs whenever a file is imported into the profiler

Reply:
This code runs (in a web worker) when the user uploads a profile JSON to the scheduling profiler.